### PR TITLE
Fix opening in new tab

### PR DESCRIPTION
**Pull request by mdbenjam**
Connects: #1714 

Opening a document in a new tab was broken. This PR fixes three things:

1. Middle clicking on a doc type will open the document in a new tab.
1. When in a new tab, we do not load the rest of the case file, and do not provide a back to file button.
1. There are tests to find regressions.

### CODE REVIEW

File: spec/feature/reader_spec.rb [hunk 1]
```diff
@@ -134,7 +134,7 @@ def add_comment(text)
       scenario "Enter a case" do
         visit "/reader/appeal"
 
-        expect(page).to have_content(appeal.veteran_last_name)
+        expect(page).to have_content(appeal.veteran_full_name)
```
Comment (mdbenjam) [inline spec/feature/reader_spec.rb]: Unrelated test fix.

**Comment by tejans24** [comment]
I ran the added test a few times, it seems to be passing consistently.

File: spec/feature/reader_spec.rb [hunk 2]
```diff
@@ -143,6 +143,27 @@ def add_comment(text)
...
     end
 
+    scenario "Open document in new tab" do
+      # Open the URL that the first document button points to. We cannot simply
+      # click on the link since we've overridden the mouseup event to not open
+      # the link, but instead to move to the document view in the SPA. Middle clicking
+      # is not overridden, but I cannot figure out how to middle click in the test.
+      # Instead we just visit the page specified by the link.
+      visit "/reader/appeal/#{appeal.vacols_id}/documents"
+      single_link = find_link(documents[0].type)[:href]
```
Comment (tejans24) [inline spec/feature/reader_spec.rb]: if middle isn't possible, can we try the control/command + click on the link instead?

It takes it to the same link but I feel like that would cover the feature a bit more.

File: spec/feature/establish_claim_spec.rb [hunk 3]
```diff
@@ -522,7 +522,7 @@
           ]
         end
 
-        scenario "Assigning it to complete the claims establishment" do
+        skip "Assigning it to complete the claims establishment" do
```
Comment (tejans24) [inline spec/feature/establish_claim_spec.rb]: 👍

File: client/app/components/Link.jsx [hunk 4]
```diff
@@ -5,13 +5,18 @@ import { Link as RouterLink } from 'react-router-dom';
 const CLASS_NAME_MAPPING = {
   primary: 'usa-button',
   secondary: 'usa-button-outline',
-  disabled: 'usa-button-disabled'
+  disabled: 'usa-button-disabled',
```
Comment (tejans24) [inline client/app/components/Link.jsx]: 👍

File: spec/feature/reader_spec.rb (hunk 2 above)
Comment (mdbenjam) [inline spec/feature/reader_spec.rb]: When I tried to actually open a new tab and test, it made the tests unstable. I'm also not sure that we should be testing that links work. Instead, I've added some UI tests to fix make sure that we follow the links. Let me know what you think.

File: client/test/node/reader/DecisionReviewer-test.js [hunk 5]
```diff
@@ -214,6 +215,69 @@ describe('DecisionReviewer', () => {
   context('PDF list view', () => {
     beforeEach(() => setUpDocuments());
 
+    context('follows the link when', () => {
+      it('document is ctrl + clicked', asyncTest(async() => {
```
Comment (mdbenjam) [inline client/test/node/reader/DecisionReviewer-test.js]: I'm not sure how I feel about these tests. It seems difficult to make sure that we allow links to be followed? Open to other ideas here.
